chore(releasing): Prepare v0.40.0 release (#20924)
vectordotdev/vector: website/content/en/highlights/2024-07-29-0-40-0-upgrade-guide.md

---
date: "2024-07-29"
title: "0.40 Upgrade Guide"
description: "An upgrade guide that addresses breaking changes in 0.40.0"
authors: ["jszwedko"]
release: "0.40.0"
hide_on_release_notes: false
badges:
  type: breaking change
---

Vector's 0.40.0 release includes three **breaking changes**:

1. [GELF codec defaults to null-delimited messages for stream-based sources](#gelf)
1. [Reduce transforms now properly aggregate nested fields](#reduce-aggregate)
1. [Vector no longer supports CentOS 7](#centos7)

> We cover them below to help you upgrade quickly:

Reviewer: ```suggestion
The changes are covered below to help you upgrade quickly:
```
